File: app/src/main/java/com/karla/learningverbs/view/profile/GetPasswordActivity.java
package com.karla.learningverbs.view.profile;

import static com.karla.learningverbs.utils.StringUtils.validateEmail;

import androidx.annotation.NonNull;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Toast;

import com.karla.learningverbs.R;
import com.karla.learningverbs.databinding.ActivityGetPasswordBinding;
import com.karla.learningverbs.viewmodel.GetPasswordViewModel;
import com.karla.learningverbs.utils.base.BaseActivity;

public class GetPasswordActivity extends BaseActivity<ActivityGetPasswordBinding, GetPasswordViewModel> {

    @Override
    protected GetPasswordViewModel createViewModel() {
        return new ViewModelProvider(this).get(GetPasswordViewModel.class);
    }

    @NonNull
    @Override
    protected ActivityGetPasswordBinding createViewBinding(LayoutInflater layoutInflater) {
        return ActivityGetPasswordBinding.inflate(layoutInflater);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding.btnGetPassword.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                validate();
            }
        });

    }

    public void validate() {
        String email = binding.editTextEmail.getText().toString().trim();
        if (email.isEmpty() || !validateEmail(binding.editTextEmail.getText().toString())) {
            binding.editTextEmail.setError(getString(R.string.msg_no_email_send));
        } else {
            binding.editTextEmail.setError(null);
            viewModel.getRecoveryPassword(binding.editTextEmail.getText().toString());
        }

        recoverPassword();
    }

    private void recoverPassword() {
        viewModel.sendEmail().observe(GetPasswordActivity.this, new Observer<String>() {
            @Override
            public void onChanged(String string) {
                if (string != null) {
                    Toast.makeText(GetPasswordActivity.this, string, Toast.LENGTH_SHORT).show();
                    onBackPressed();
                }
            }
        });
    }

}

File: app/src/main/java/com/karla/learningverbs/kotlin/view/profile/GetPasswordActivity.kt
package com.karla.learningverbs.kotlin.view.profile

import android.os.Bundle
import android.view.LayoutInflater
import android.widget.Toast
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.karla.learningverbs.R
import com.karla.learningverbs.databinding.ActivityGetPasswordBinding
import com.karla.learningverbs.kotlin.utils.base.BaseActivity
import com.karla.learningverbs.kotlin.viewmodel.GetPasswordViewModel
import com.karla.learningverbs.kotlin.utils.StringUtils

class GetPasswordActivity : BaseActivity<ActivityGetPasswordBinding, GetPasswordViewModel>() {

    override fun createViewModel(): GetPasswordViewModel {
        return ViewModelProvider(this).get(GetPasswordViewModel::class.java)
    }

    override fun createViewBinding(layoutInflater: LayoutInflater?): ActivityGetPasswordBinding{
        return ActivityGetPasswordBinding.inflate(layoutInflater!!)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding!!.btnGetPassword.setOnClickListener { validate() }
    }

    fun validate() {
        val email = binding!!.editTextEmail.text.toString().trim { it <= ' ' }
        if (email.isEmpty() || !StringUtils.validateEmail(binding!!.editTextEmail.text.toString())) {
            binding!!.editTextEmail.error = getString(R.string.msg_no_email_send)
        } else {
            binding!!.editTextEmail.error = null
            viewModel!!.getRecoveryPassword(binding!!.editTextEmail.text.toString())
        }
        recoverPassword()
    }

    private fun recoverPassword() {
        viewModel?.sendEmail()?.observe(this@GetPasswordActivity, Observer { string ->
            if (string != null) {
                Toast.makeText(this@GetPasswordActivity, string, Toast.LENGTH_SHORT).show()
                onBackPressed()
            }
        })
    }

}